arangod/Agency/AgencyComm.cpp

```diff
@@ -646,14 +625,14 @@ AgencyCommResult AgencyComm::sendServerState(double timeout) {
                           builder.slice(), 0, timeout));
 }
 
-std::string AgencyComm::version() {
+std::string_view AgencyComm::version() {
   AgencyCommResult result =
       sendWithFailover(arangodb::rest::RequestType::GET,
                        AgencyCommHelper::CONNECTION_OPTIONS._requestTimeout,
                        "/_api/version", VPackSlice::noneSlice());
 
-  if (result.successful()) {
-    return result._body;
+  if (result.successful() && result.slice().isString()) {
+    return result.slice().stringView();
```

Comment:
if `result` is a local variable here and goes out of scope directly after the `return` statement, isn't unsafe to return a `string_view` from here that points into the internals of `result`?
I would like this API to return a self-contained value as it was the case before. Otherwise it seems to be unsafe.

Reply:
Right, thanks!